src/material/datepicker/date-range-input-parts.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, ElementRef, Optional, Self, InjectionToken, Inject} from '@angular/core';
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  AbstractControl,
+  ValidationErrors,
+  NgForm,
+  FormGroupDirective,
+  NgControl,
+} from '@angular/forms';
+import {
+  CanUpdateErrorState,
+  CanDisable,
+  ErrorStateMatcher,
+  CanDisableCtor,
+  CanUpdateErrorStateCtor,
+  mixinErrorState,
+  mixinDisabled,
+} from '@angular/material/core';
+import {BooleanInput} from '@angular/cdk/coercion';
+
+/** Parent component that should be wrapped around `MatStartDate` and `MatEndDate`. */
```

Comment:
Would it be accurate to say this?
```suggestion
/** Container component around `MatStartDate` and `MatEndDate` that coordinates changes. */
```